### DISCUSSION TITLE
chart-fx most active compared to its alternatives? Try to realize simple real-time candlesticks, any alternative is fine.

### DISCUSSION THREAD

**Discussion by yezhengli-Mr9** [GSI-CS-CO/chart-fx#644] [General]
By checking,chart-fx most active compared to its alternatives? Try to realize simple real-time candlesticks #641  (or #643) #642 , any alternative is fine. 
> Chart performance comparison for popular JavaFX, Java-Swing, C++/Qt and WebAssembly-based implementations: <a href="https://github.com/extjfx/extjfx">ExtJFX</a>, <a href="https://github.com/fair-acc/chart-fx">ChartFx</a>, <a href="https://github.com/HanSolo/charts">HanSolo Charts</a>, <a href="http://www.jfree.org/jfreechart/">JFreeChart</a>, <a href="https://cds.cern.ch/record/1215878">JDataViewer</a>, <a href="https://www.qcustomplot.com/">QCustomPlot</a>, <a href="https://doc.qt.io/qt-5/qtcharts-index.html">Qt-Charts</a>, <a href="https://doc.qt.io/qt-5/wasm.html">WebAssembly</a>. The last `Qt Charts` entries show results for 100k data points being updated at 25 Hz
0. not considering `python` or `R` right now since feel they are for batch data analysis mostly.
1. <a href="https://github.com/extjfx/extjfx">ExtJFX</a> last updated on [Mar 01st 2019](https://github.com/extjfx/extjfx/commit/ec4d54e5cb20bbc62a021d2897a2fd1210bb923f).
2. <a href="https://github.com/HanSolo/charts">HanSolo Charts</a> last updated on [Dec. 13th 2022](https://github.com/HanSolo/charts/commit/f9210ad4b08d52519be087c7aaae5108da6de1d8)
3. <a href="http://www.jfree.org/jfreechart/">JFreeChart</a> last updated on [Jan. 08th 2023](https://github.com/jfree/jfreechart)
4. <a href="https://cds.cern.ch/record/1215878">JDataViewer</a>, even official release page was last modified on 2019-02-21.
 ![image](https://github.com/fair-acc/chart-fx/assets/16505983/d2e76590-3d04-41b8-bb88-29630eb923f9)
5. <a href="https://www.qcustomplot.com/">QCustomPlot</a> c++
6. <a href="https://doc.qt.io/qt-5/qtcharts-index.html">Qt-Charts</a> c++
7. <a href="https://doc.qt.io/qt-5/wasm.html">WebAssembly</a> c++

**Comment by raven2cz** [comment]
I have fairly extensive personal business frameworks, and I thought I would dedicate at least a substantial part of the foundation to open source. Java and JavaFX are closest to my internal libraries, so chart-fx was a good choice, unfortunately it's not exactly tailored to the business style, rather for measuring purposes, but their optimization for rendering speed and the way some renderers are processed is very close to me. There weren't many options in this field.

The C++ language is not much of an option anymore, in terms of the possibility of memory leaks, especially in the open source environment, they seem very dangerous to me, especially for business purposes. I would rather choose rust, but just a few years ago it was more of a dream than reality.

However, I have not finished my work, I am missing the adoption of volume profiles and foot charts, and some adjustments, especially of time segments, which chart-fx does not yet have.

In the end, however, it is only about the rendering part, not about taking over the execution platform or computing platform. Although I "sampled" a fairly huge part in tests for display, especially real-time processing for numbers bars.

**Reply by yezhengli-Mr9** [reply]
Thanks @raven2cz, I am for personal use and in short term will stick to java (more friendly for cross-platforms, macbook, linux than, at least, c/c++); 
* Speaking of "real-time", 1-minute delay is fine to me. It suffices for me to use bars within one day, that is at most 60*24= 1,440 bars so far. Do not think I will look over more than five business days in the short term.
* at least timestamp conversion and alerts setups in java are relatively doable in `java`;
* indeed used to consider c++, but thanks for your professional suggestion (then to give up c++ in short term; also remember Qt4/5 takes long to install on macbook then? -- correct me if I am wrong). 
* can consider rust later.

**Volume profiles**: volume profiles are beneficial for me in long term #294 while short-term I am to put effort in stop losses #641 in terms of regression testing (for forecasting) and visualization --visualization seems important for insights and backtesting.

**Foot chart**: I do not have API for historical footprint chart, only streaming footprint data; consequentially, personally not considering foot charts so far.

**Reply by yezhengli-Mr9** [reply]
Might be interested in volume profiles after finishing some fundamental prototypes on my end~

**Reply by yezhengli-Mr9** [reply]
# adoption of volume profiles

~nterested in generating on right side a projection chart of volume (delta) in the upcoming weeks for~
https://github.com/fair-acc/chart-fx/blob/ed86318c5a1bde23d285633a94860648aee5fb32/chartfx-samples/src/main/java/io/fair_acc/sample/financial/FinancialRealtimeCandlestickSample.java#L49
~although I am still resolving #647 at least for this week.
~priority seems lower than **volume delta and foot(print) charts (better in relatively real time)**.~

**Reply by yezhengli-Mr9** [reply]
#  volume delta and foot(print) charts (better in relatively real time)
Interested in volume delta and foot(print) charts (better in relatively real time, interested in observing their values for several segments/ time intervals within each day-- for example, 3-5 segments/time intervals within each day) in upcoming weeks, so feel still have to fix #647 first.
~Put on hold due to algorithm's logic change.~
